### PR TITLE
reorder_vertices for 1 and 2d #6207

### PR DESCRIPTION
**Pull request by victor-dr-z**
1. Added Triangulation<1,1> and <2,2>::reorder_vertices
2. Test cases added
3. Doc/news/minor added

### CODE REVIEW

File: source/grid/tria.cc [hunk 1]
```diff
@@ -9221,6 +9222,49 @@ create_triangulation_compatibility (const std::vector<Point<spacedim> > &v,
...
+    std::swap(points[0], points[1]);
+}
+
+template <>
+void
+Triangulation<2,2>::
+reorder_vertices (std::vector<Point<2>> &points)
+{
+  Assert(points.size()==GeometryInfo<2>::vertices_per_cell,
+         ExcMessage("Number of vertices does not match GeometryInfo<2>::vertices_per_cell"));
```
Comment (masterleinad) [inline source/grid/tria.cc]: Just use `AssertDimension`.

File: source/grid/tria.cc [hunk 2]
```diff
@@ -9221,6 +9222,49 @@ create_triangulation_compatibility (const std::vector<Point<spacedim> > &v,
 }
 
 
+template <>
+void
+Triangulation<1,1>::
+reorder_vertices (std::vector<Point<1>> &points)
+{
+  Assert(points.size()==GeometryInfo<1>::vertices_per_cell,
+         ExcMessage("Number of vertices does not match GeometryInfo<1>::vertices_per_cell"));
```
Comment (masterleinad) [inline source/grid/tria.cc]: Use `AssertDimension`.

File: tests/grid/tria_reorder_vertices.cc [hunk 3]
```diff
@@ -0,0 +1,69 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2017 by the deal.II authors
```
Comment (masterleinad) [inline tests/grid/tria_reorder_vertices.cc]: Just use
```
// Copyright (C) 2018 by the deal.II authors
```

**Comment by bangerth** [comment]
@weixiong-zheng-berkeley: Any news?

**Comment by weixiong-zheng-berkeley** [comment]
Sorry. The past whole month was for moving in CA and moving my home totally from TX to CA and just finished. I will try to restart from where I dropped off ASAP.

**Comment by masterleinad** [comment]
Please rebase this PR onto the current master soon to make sure this PR is conforming to the new indenting style we are using. If you need to help, don't hesitate to ask.

**Comment by bangerth** [comment]
@weixiong-zheng-berkeley -- ping?

**Comment by bangerth** [comment]
@weixiong-zheng-berkeley -- any news?

**Comment by drwells** [comment]
@victor-dr-z since this is still WIP lets move the milestone to 9.2

**Comment by drwells** [comment]
This is a very specialized function - we are usually interested in the ordering of the FE space, not of the vertices themselves. Unless someone is interested enough in this functionality to finish this PR I suggest we close this.

**Comment by bangerth** [comment]
I agree. We can always re-open the PR if anyone has a need for this functionality -- though it's probably true that there is little given that we've heard this kind of request only once in 20 years.

**Pull request closed without merging**